### PR TITLE
Design Experiment: Form Recognizer custom deserialization

### PR DESCRIPTION
**Pull request by annelo-msft**
This Draft PR is an early investigation into the following areas:

- Strongly typing prebuilt models: https://github.com/Azure/azure-sdk-for-net/issues/16330
- Using patterns developed in the [Search API](https://apiview.dev/Assemblies/Review/d40a51d26cf2444a85d302758ca3a2d1#Azure.Search.Documents.SearchClient) for [custom serialization of static types](https://github.com/Azure/azure-sdk/issues/997).

### CODE REVIEW

File: sdk/formrecognizer/Azure.AI.FormRecognizer/Azure.AI.FormRecognizer.sln [hunk 1]
```diff
@@ -7,7 +7,9 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Azure.AI.FormRecognizer", "
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Azure.AI.FormRecognizer.Tests", "tests\Azure.AI.FormRecognizer.Tests.csproj", "{4F476D56-DDE7-43D3-8CB4-BA1E77F5A300}"
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Azure.Core.TestFramework", "..\..\core\Azure.Core.TestFramework\src\Azure.Core.TestFramework.csproj", "{439F1494-8E96-4931-AB0A-5BBA7EBA15D2}"
```
Comment (annelo-msft) [inline sdk/formrecognizer/Azure.AI.FormRecognizer/Azure.AI.FormRecognizer.sln:10]: Changes in this file are only to add a sample project for illustrating purposes and will never be merged into master.

File: sdk/formrecognizer/Azure.AI.FormRecognizer/CustomSerializationSample/CustomSerializationSample.csproj [hunk 2]
```diff
@@ -0,0 +1,16 @@
+<Project Sdk="Microsoft.NET.Sdk">
```
Comment (annelo-msft) [inline sdk/formrecognizer/Azure.AI.FormRecognizer/CustomSerializationSample/CustomSerializationSample.csproj:1]: This project won't be merged - it's just a super simple console app to hold samples.

**Comment by jsquire** [comment]
Hi @annelo-msft.  There hasn't been recent engagement on this PR.  If this is still an active work stream, please let us know by removing the `no-recent-activity` label.  Otherwise, we'll close this out in 7 days.

**Comment by maririos** [comment]
Thanks @jsquire we have linked this PR into issues so we can close it now. Thank you!

**Pull request closed without merging**